charts/spire/charts/spire-agent/templates/_helpers.tpl

```diff
@@ -72,3 +72,7 @@ Create the name of the service account to use
 {{- printf "%s/%s" .image.registry .image.repository -}}
 {{- end -}}
 {{- end }}
+
+{{- define "spire-server-service" -}}
+{{ include "spire-agent.fullname" . | trimSuffix "-agent" }}-server
```

Comment:
This goes okay by convention of the overall chart. If people use this subchart directly, this could fail. 
OK for now, we can assume that this is always used as subchart.

Reply:
That is why it is build as subchart and not as a separate chart. In the registry only the top level chart is published.